pymatgen/io/cp2k/inputs.py

```diff
@@ -0,0 +1,1364 @@
+    For example, the section "FORCE_EVAL" activates the evaluation of forces/energies, but within
+    "FORCE_EVAL" the keyword "METHOD" controls whether or not this will be done with, say,
+    "Quickstep" (DFT) or "EIP" (empirical interatomic potential).
+    """
+
+    def __init__(
+        self,
+        name,
+        *args,
+        description=None,
+        units=None,
+        verbose=True,
+        repeats=False
+    ):
+        """
+        Initializes a keyword. These Keywords and the value passed to them are sometimes as simple as KEYWORD VALUE,
+        but can also be more elaborate such as KEYWORD [UNITS] VALUE1 VALUE2, which is why this class exists:
+        to handle many values and control easy printing to an input file.
+
+        Args:
+            name (str): The name of this keyword. Must match an acceptable keyword from CP2K
+            args: All non-keyword arguments after 'name' are interpreted as the values to set for
+                this keyword. i.e: KEYWORD ARG1 ARG2 would provide to values to the keyword.
+            description (str): The description for this keyword. This can make readability of
+                input files easier for some. Default=None.
+            units (str): The units for this keyword. If not specified, CP2K default units will be
+                used. Consult manual for default units. Default=None.
+            repeats (bool): Whether or not this keyword may be repeated. Default=False.
+        """
+        self.name = name
+        self.values = [a for a in args]
+        self.description = description
+        self.repeats = repeats
+        self.units = units
+        self.verbose = verbose
+
+    def __str__(self):
+        return (
+            self.name.__str__()
+            + " "
+            + ("[{}] ".format(self.units) if self.units else "")
+            + " ".join(map(str, self.values))
+            + (
+                " ! " + self.description
+                if (self.description and self.verbose)
+                else ""
+            )
+        )
+
+    def __eq__(self, other):
+        if self.name == other.name:
+            if self.values == other.values:
+                if self.units == self.units:
+                    return True
+        return False
+
+    def as_dict(self):
+        """
+        Get a dictionary representation of the Keyword
+        """
+        d = {}
+        d["name"] = self.name
+        d["values"] = self.values
+        d["description"] = self.description
+        d["repeats"] = self.repeats
+        d["units"] = self.units
+        d["verbose"] = self.verbose
+        return d
+
+    @classmethod
+    def from_dict(cls, d):
+        """
+        Initialise from dictonary
+        """
+        return Keyword(
+            d["name"],
+            *d["values"],
+            description=d["description"],
+            repeats=d["repeats"],
+            units=d["units"],
+            verbose=d["verbose"]
+        )
```

Comment:
So when I do it with`MSONable`, I don't know if I can keep *args, because `MSONable` wants everything to be a key value pair, and `*args` only becomes key value when it is interpreted as a list by `self.values = [a for a in args]`. 

If I make it MSONable and have `self.values = [v for v in *values]`, as_dict() doesn't keep the variable self.values because it wasn't a key-value pair in the arguments.

Now I can change it and instead of doing *args have `values=[]` as one of the arguments to be the list of values that go with the Keyword. It just doesn't look as clean I think.

Does that make sense?

Reply:
Oh! Can you save the list under both `self.args` and `self.values`

Reply:
Actually no, that won't work either. You'll need to have the init argument as `args` as well, not `values`

Reply:
``` python

def __init__(self, ..... , *args):
    self.args = args
    self.values = args
```

Reply:
Yeah. So I think it makes sense if we want to keep all the serialization done by monty to rewrite it as `__init__(name, values=[], ...)`. If we want it to be maximally intuitive for someone, I think being able to write Keyword('my_keyword', 'value1', 'value2') is better.

Reply:
No my little rewrite will still let you do that. `monty` is fine with varidac args, it just has to be called `args` in the init and stored under `args`

Reply:
> ```python
> def __init__(self, ..... , *args):
>     self.args = args
>     self.values = args
> ```

Then when you do 

>```python
> k  = Keyword('keyword', 'value')
> print(k.as_dict())
>```

will give 
`{'@module': '__main__', '@class': 'Keyword', '@version': '0.2', 'name': 'keyword'}`

Reply:
Did you leave the other options in place? I'm not suggesting a reordering, just renaming;
So in init arglist: `values` -> `args`

Reply:
Nevermind, you're right. This is a problem with Monty. Let me see if I can figure that out.

Reply:
I saw you merged into Monty, *but* it won't work for a reason I didn't realize... 

When serialize the *args with Monty into a dict you'll get something like {'arg': [0,1,2]}. Now, when we use from_dict, there is no way of knowing that the 'arg' was part of *args rather than **kwargs, because they both get stored as a key-value pair. So monty will have to interpret it is a kwarg and will throw an error because it is an unknown keyword.